Small cleanup of EventTargetFlag

rniwa opened the pull request:
#### c4b49d2e4b5cf0a30d961409f4a254ac20651bda
<pre>
Small cleanup of EventTargetFlag
<a href="https://bugs.webkit.org/show_bug.cgi?id=244902">https://bugs.webkit.org/show_bug.cgi?id=244902</a>

Reviewed by Chris Dumez and Yusuke Suzuki.

* Source/WebCore/dom/EventTarget.h:
(WebCore::EventTarget::hasEventTargetData const):
(WebCore::EventTarget::isNode const):
(WebCore::EventTarget::EventTarget):
(WebCore::EventTarget::setHasEventTargetData const): Deleted.
(WebCore::EventTarget::ensureEventTargetData):
(WebCore::EventTarget::hasEventTargetFlag const): Added.
(WebCore::EventTarget::setEventTargetFlag): Added.

Canonical link: <a href="https://commits.webkit.org/254252@main">https://commits.webkit.org/254252@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2bb2544e40fad0ed9361783a00679d2b5ba8b5f0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/88498 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33029 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19366 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/97674 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/153153 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/92460 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/31523 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/27120 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/80693 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92328 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94101 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25027 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/75414 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/24988 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/79914 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/80013 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/67951 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/29141 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/14003 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/67/builds/29104 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15021 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2992 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/32372 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/37947 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31178 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34130 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/dom/EventTarget.h:
```diff
@@ -172,32 +165,33 @@ class EventTarget : public ScriptWrappable, public CanMakeWeakPtr<EventTarget, W
     virtual void eventListenersDidChange() { }
 
+    bool hasEventTargetFlag(EventTargetFlag flag) const { return weakPtrFactory().bitfield() & static_cast<uint16_t>(flag); }
+    void setEventTargetFlag(EventTargetFlag flag, bool value)
```
Given the number of lines, I think we should move it out of the class and mark it as inline instead of inlining it in the class.

Constellation reviewed (approved):
r=me

rniwa committed:
Small cleanup of EventTargetFlag
https://bugs.webkit.org/show_bug.cgi?id=244902

Reviewed by Chris Dumez and Yusuke Suzuki.

* Source/WebCore/dom/EventTarget.h:
(WebCore::EventTarget::hasEventTargetData const):
(WebCore::EventTarget::isNode const):
(WebCore::EventTarget::EventTarget):
(WebCore::EventTarget::setHasEventTargetData const): Deleted.
(WebCore::EventTarget::ensureEventTargetData):
(WebCore::EventTarget::hasEventTargetFlag const): Added.
(WebCore::EventTarget::setEventTargetFlag): Added.

Canonical link: https://commits.webkit.org/254252@main
